Experimental revamp of get-started - a more user friendly version

Ju-221 opened the pull request:
This script enhancement can be run from get-started-exp.bat. It consolidates all the .bat files into one single script along with a integrity check and checks the installation of python is done properly along with a smarter IPV4 selection to determine the correct internal IP based on the gateway and if the interface is active. 

Still in experimental state - might help users install the repo faster. The script downloads python 3.8.0 if it's not installed, as well checks if the master folder has it previously downloaded. I would suggest highly to bundle the binary VC_redist.x64.exe and python-3.8.0-amd64.exe in the release package if possible. The script doesn't require any binaries however to run.

Ju-221 committed:
Experimental - GetStarted revamp

A small revamp of the original Get-started along with other information about getStarted.bat along with all the other.bat files and consolidated it into one single powershell script that checks the integirty of the project on runtime and has a more advanced getIPAddress function that determines the internal IP based on Private network space and if the gateway is available among other small stuff.

Ju-221 committed:
Experminetal- GetStarted script

Experminetal- GetStarted script: A more powerful version of the .bat files in the master branch, allows to check integrity with the project at runtime and makes it foolproof for users to install the repo. 

Not tested thoroughly yet.

Ju-221 committed:
Update get-started-exp.ps1

Added experimental 32bit support to the script. Not tested with a 32bit machine yet. Will add in the following revision arguments to skip checking the python installation if it becomes a problem

kavezo commented:
hmm, tried running it on my sister's computer, I ended up with the following:

```
File C:\some\filepath\ZipZap-master\get-started-exp.ps1 cannot be loaded because running scripts is 
disabled on this system. For more information, see about_Execution_Policies at https:/go.microsoft.com/fwlink/?LinkID=135170.
    + CategoryInfo          : SecurityError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : UnauthorizedAccess
```

Could we automate enabling script running, or is that something we have to ask the user to do?

Ju-221 commented:
Hi, you have to click on the get-started-exp.bat first so that it may authenticate it's an administrator then it runs the powershell script automatically. By default win10 does not allow direct execution of PS1 files unless the user specifies

kavezo commented:
yep, tried running the .bat file (couldn't see the error code before the PS terminal closed on its own), then tried running it in the PS IDE-thing as admin, and that's where I saw the error message

Ju-221 commented:
Which version of windows are you running? Is the user an admin?

kavezo commented:
Windows 10, user is an admin

Bypassed it manually and now I'm starting to wonder if something wrong happened in the file transfer, I just saved the raw files from github because don't want to install too much stuff like git onto my sister's computer and looking at it idk if it's supposed to be
```
[Security.Principal.WindowsBuiltInRole] â€œAdministratorâ€))
```

Ju-221 commented:
It seems some encoding problem from utf-8, the line here should be 

`    [Security.Principal.WindowsBuiltInRole] “Administrator”))`

It seems encoded as â€œAdministratorâ€� for whatever reason

Ju-221 commented:
Let me make some changes to replace the encoding correctly, might have been an error from ANSI to utf at some point. I'll edit it on the script, please try again with the .bat file that it comes with once the change is done

kavezo commented:
thank you!

Ju-221 committed:
Update get-started-exp.ps1

Updated file and tested -“ was being used instead of " which caused a decoding error in the cmd console. PS reads â€œ instead of " due to the encoding of PS by default.

Ju-221 commented:
There you go, added as well a PAUSE in the admin error so that the user can read that they don't have admin creds or if an error-handling exception happens so that they may read it. Try it now from the .bat file it should work fine, I hope.

kavezo commented:
oof this was such good work, but it's completely unapplicable with the new version, sorry...
